From: Cliff Wickman <cpw@sgi.com>
Subject: x86/UV2: Ack BAU interrupt earlier
Git-commit: 88ed9dd7f63c3ae71c1984d99ee2dced0b386dea
Patch-mainline: Yes
References: bnc#743212

This patch moves the ack of the BAU interrupt to the beginning of 
the interrupt handler so that there is less possibility of a lost
interrupt and slower response to a shootdown message.

Signed-off-by: Cliff Wickman <cpw@sgi.com>
Signed-off-by: Rafael J. Wysocki <rjw@suse.de>
---
 arch/x86/platform/uv/tlb_uv.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

Index: linux-3.0-SLE11-SP2/arch/x86/platform/uv/tlb_uv.c
===================================================================
--- linux-3.0-SLE11-SP2.orig/arch/x86/platform/uv/tlb_uv.c
+++ linux-3.0-SLE11-SP2/arch/x86/platform/uv/tlb_uv.c
@@ -1218,6 +1218,7 @@ void uv_bau_message_interrupt(struct pt_
 	struct ptc_stats *stat;
 	struct msg_desc msgdesc;
 
+	ack_APIC_irq();
 	time_start = get_cycles();
 
 	bcp = &per_cpu(bau_control, smp_processor_id());
@@ -1247,8 +1248,6 @@ void uv_bau_message_interrupt(struct pt_
 		stat->d_nomsg++;
 	else if (count > 1)
 		stat->d_multmsg++;
-
-	ack_APIC_irq();
 }
 
 /*
